docs/source/jwst/datamodels/models.rst

```diff
@@ -213,154 +155,212 @@ Searching a model
+    for entry in entries:
+      pass
 
-The ``extra_fits`` namespace may also hold entire hdus that are not
-mapped to a data model.  For example, if the FITS file contains an
-extension called ``EXTRA``, it can be accessed using::
+To add an entry to the history, first create the entry by calling
+``stdatamodels.util.create_history_entry`` and appending the entry to the model
+history::
 
-    model.extra_fits.EXTRA
+    import stdatamodels
+    entry = stdatamodels.util.create_history_entry("Processed through the frobulator step")
+    model.history.append(entry)
 
-and its data array can be accessed using::
+These history entries are stored in ``HISTORY`` keywords when saving
+to FITS format. As an option, history entries can contain a dictionary
+with a description of the software used. The dictionary must have the
+following keys:
 
-    model.extra_fits.EXTRA.data
+  * ``name``: The name of the software
+  * ``author``: The author or institution that produced the software
+  * ``homepage``: A URI to the homepage of the software
+  * ``version``: The version of the software
 
-To get a list of everything in ``extra_fits`` as a dictionary, use::
+The calling sequence to create a history entry with the software
+description is::
 
-    model.extra_fits.instance
+    entry =  stdatamodels.util.create_history_entry(description, software=software_dict)
 
-(``instance`` can be used at any node in the tree, not just ``extra_fits``,
-to return a dictionary of rest of the tree structure at that node.)
+where the second argument is the dictionary with the keywords
+mentioned.
 
-.. note::
+Working with lists
+------------------
 
-    The ``jwst`` pipeline never directly accesses information
-    from ``extra_fits``, as this would bypass the schema validation and partly defeat
-    the purpose of the data model. If you are developing a model for pipeline use, 
-    it is strongly recommended to define any new
-    (meta)data in the datamodel schema early on in the development process.
+Unlike ordinary Python lists, lists in the schema may be restricted to
+only accept a certain set of values.  Items may be added to lists in
+two ways: by passing a dictionary containing the desired key/value
+pairs for the object, or using the lists special method ``item`` to
+create a metadata object and then assigning that to the list.
+
+For example, suppose the metadata element ``meta.transformations`` is a
+list of transformation objects, each of which has a ``type`` (string)
+and a ``coeff`` (number) member.  We can assign elements to the list in
+the following equivalent ways::
+
+    trans = model.meta.transformations.item()
+    trans.type = 'SIN'
+    trans.coeff = 42.0
+    model.meta.transformations.append(trans)
+    model.meta.transformations.append({'type': 'SIN', 'coeff': 42.0})
+
+When accessing the items of the list, the result is a normal metadata
+object where the attributes are type-checked::
+  
+    >>> trans = model.meta.transformations[0]
```

Comment:
You might actually want `.. doctest-skip::` block here since `>>>` is needed to distinguish code from output, and I don't think the doctest will pass.